packages/file-search/src/browser/quick-editors-open.ts

```diff
@@ -0,0 +1,176 @@
+export class QuickEditorsOpenService implements QuickOpenModel, QuickOpenHandler {
+
+    @inject(ApplicationShell)
+    protected readonly applicationShell: ApplicationShell;
+
+    @inject(FileSystem)
+    protected readonly fileSystem: FileSystem;
+
+    @inject(WorkspaceService)
+    protected readonly workspaceService: WorkspaceService;
+
+    @inject(OpenerService)
+    protected readonly openerService: OpenerService;
+
+    @inject(PrefixQuickOpenService)
+    protected readonly prefixQuickOpenService: PrefixQuickOpenService;
+
+    @inject(MessageService)
+    protected readonly messageService: MessageService;
+
+    @inject(LabelProvider)
+    protected readonly labelProvider: LabelProvider;
+
+    readonly prefix: string = 'edt ';
+
+    get description(): string {
+        return 'Show All Opened Editors';
+    }
+
+    getModel(): QuickOpenModel {
+        return this;
+    }
+
+    getOptions(): QuickOpenOptions {
+        return {
+            placeholder: '',
+            fuzzyMatchLabel: {
+                enableSeparateSubstringMatching: true
+            },
+            fuzzyMatchDescription: {
+                enableSeparateSubstringMatching: true
+            },
+            showItemsWithoutHighlight: true,
+            onClose: () => {
+                this.cancelIndicator.cancel();
+            }
+        };
+    }
+
+    isEnabled(): boolean {
+        return this.workspaceService.opened;
+    }
+
+    open(): void {
+        this.prefixQuickOpenService.open(this.prefix);
+    }
+
+    private cancelIndicator = new CancellationTokenSource();
+
+    public async onType(lookFor: string, acceptor: (items: QuickOpenItem[]) => void): Promise<void> {
+        this.cancelIndicator.cancel();
+        this.cancelIndicator = new CancellationTokenSource();
+        const token = this.cancelIndicator.token;
+
+        const editorItems: QuickOpenItem[] = [];
+
+        // Get URIs of all EditorWidgets that are currently opened
+        // Sorted by URI
+        const widgets = this.applicationShell.widgets
+            .filter((widget): widget is EditorWidget => (widget instanceof EditorWidget))
+            .map(w => w.getResourceUri()!)
+            .sort();
+
+        if (widgets.length === 0) {
+            editorItems.push(new QuickOpenItem({
+                label: 'List of opened editors is currently empty',
+                run: () => false
+            }));
+            acceptor(editorItems);
+            return;
+        }
+
+        for (const uri of widgets) {
+            if (uri.scheme === 'file' && fuzzy.test(lookFor, uri.toString())) {
+                const item = await this.toItem(uri);
+                if (!!token.isCancellationRequested) {
+                    return;
+                }
+                editorItems.push(item);
+            }
+            acceptor(editorItems);
+        }
+        return;
+    }
+
+    private async toItem(uriOrString: URI | string, group?: QuickOpenGroupItemOptions) {
```

Comment:
This method is `private` and you are only ever passing a `uri` so it's enough for the signature to be the following:

```ts
private async toItem(uri: URI, group?: QuickOpenGroupItemOptions): Promise<QuickOpenItem<QuickOpenItemOptions>>
```

Also, please make sure to add the `return` type for methods.